### DISCUSSION TITLE
Issue with creating  a Language object from a loaded config.

### DISCUSSION THREAD

**Discussion by Vadim-Maklakov** [explosion/spaCy#9072] [Help: Coding & Implementations] [answered]
Hi all!
I check the training model with CLI using  ner_demo project from yours projects folder – model training and predict after loading from HDD very well.
Now I am trying to train the model in Spyder IDE using Language.from_config(config) instead of using CLI.
https://spacy.io/api/language#from_config. 
code:
```
import spacy
from thinc.api import Config
from spacy.language import Language
#Import config setting for train inside Spyder
config = Config().from_disk("/home/mvg/Ewps/Spacy3/config.cfg")
nlp = Language.from_config(config)
```
output from Spyder console:
```
ValueError: [E958] Language code defined in config (en) does not match language code of current
Language subclass Language (None). If you want to create an nlp object from a config, make sure 
to use the matching subclass with the language-specific settings and data.
```
I tried change in  `[nlp]` section on  ….`/site-packages/spacy/default_config.cfg`
from `lang = null` to `lang = “en”` - but on the other cases I get same  ValueError: [E958]

## Your Environment

* Operating System: Linux Debuan 10
* Python Version Used: Python 3.8.7
* spaCy Version Used: spaCy version    3.1.2  
* Environment Information: config.cfg generated with this command from spacy without any errors
`python -m spacy init config config.cfg --lang en --pipeline transformer,morphologizer,parser,ner,textcat,tagger --optimize accuracy --gpu`
and attachment to this post.
[config.cfg.zip](https://github.com/explosion/spaCy/files/7066142/config.cfg.zip)

**Comment by Vadim-Maklakov** [comment]
Hi all!
Please close this issue and please write somewhere  -  in the API or  doc string for `default_config.cfg` -  that for creating `nlp` instance from custom `config.cfg` you must have write those sentences - a) import language  b)use name language for creating `nlp` instance:
```
from spacy.lang.en import  English
config = Config().from_disk("path_to/config.cfg")
nlp = English.from_config(config)
```

**Reply by Norky101** [reply]
Hi @Vadim-Maklakov I am in the same situation as you were. How did you fix this please?

**Comment by adrianeboyd** [comment, marked as answer]
In general, we strongly recommend using the train CLI because there can be a lot of details to get right in the training loop and it manages a lot of them for you. If you don't want the use the CLI part of the equation, you should still consider using (or at least taking a careful look at) the `train_while_improving` method from `spacy.training.loop`.

If you are in doubt when loading manually like this, step through what happens in `spacy.load` in the code to understand what the required steps are here.
